executor/archive.go

```diff
@@ -67,16 +73,12 @@ func (a *ArchiveExecutor) startArchive(job *batchv1.Job, archive *k8upv1alpha1.A
 	if err != nil {
 		if !apierrors.IsAlreadyExists(err) {
 			a.Log.Error(err, "could not create job")
+			a.setConditionFalse(ConditionJobCreated, "could not create job")
```

Comment:
Please include the error into the condition message.

Having an error log followed by a condition is repeating pattern. It is probably worth the create a function to simplify this. This would also address the issue of having the message string twice. Maybe move the log into `setConditionFalse()`.

If you omit the abstraction, at least eliminate the duplication of the message strings (local variable/file level variable/constant).

Reply:
I've changed the signature of setConditonFalse() slightly to allow passing the actual errors. _Logging_ and _updating the Conditions_ are two different concerns in my opinion and that's why I would not want to mix them in one method.